### ISSUE TITLE
[Test Failure] Swig RunPlan != can't compare with None?

### ISSUE THREAD

**Issue by Robadob** [FLAMEGPU/FLAMEGPU2#1233]
Couple of Python test failures (debug/windows) whilst investigating the SWIG 4.2.1 issue, note this is with the Python wheel built with SWIG 4.2.1. Although, I've built it with 4.0.2 and received the same failures.

Not immediately clear their cause, these weren't touched in the recent bugfix so that's a coincidence.

```
====================================================== FAILURES =======================================================
____________________________________________ TestRunPlan.test_constructor _____________________________________________

self = <test_RunPlan.TestRunPlan testMethod=test_constructor>

    def test_constructor(self):
        # Create a model
        model = pyflamegpu.ModelDescription("test")
        plan = None
        plan = pyflamegpu.RunPlan(model)
>       assert plan != None

simulation\test_RunPlan.py:13:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyflamegpu.pyflamegpu.RunPlan; proxy of <Swig Object of type 'flamegpu::RunPlan *' at 0x0000022AD963CB40> >
rhs = None

    def __ne__(self, rhs):
>       return _pyflamegpu.RunPlan___ne__(self, rhs)
E       ValueError: invalid null reference in method 'RunPlan___ne__', argument 2 of type 'flamegpu::RunPlan const &'

..\..\build\lib\Debug\python\venv\Lib\site-packages\pyflamegpu\pyflamegpu.py:16808: ValueError
_________________________________________ TestRunPlanVector.test_constructor __________________________________________

self = <test_RunPlanVector.TestRunPlanVector testMethod=test_constructor>

    def test_constructor(self):
        # Create a model
        model = pyflamegpu.ModelDescription("test")
        # Declare a pointer
        plans = None
        # Use New
        initialLength = 4
        plans = pyflamegpu.RunPlanVector(model, initialLength)
>       assert plans != None

simulation\test_RunPlanVector.py:26:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyflamegpu.pyflamegpu.RunPlanVector; proxy of <Swig Object of type 'flamegpu::RunPlanVector *' at 0x0000022AD95E5BC0> >
rhs = None

    def __ne__(self, rhs):
>       return _pyflamegpu.RunPlanVector___ne__(self, rhs)
E       ValueError: invalid null reference in method 'RunPlanVector___ne__', argument 2 of type 'flamegpu::RunPlanVector const &'

..\..\build\lib\Debug\python\venv\Lib\site-packages\pyflamegpu\pyflamegpu.py:18426: ValueError
=============================================== short test summary info ===============================================
FAILED simulation/test_RunPlan.py::TestRunPlan::test_constructor - ValueError: invalid null reference in method 'RunPlan___ne__', argument 2 of type 'flamegpu::RunPlan const &'
FAILED simulation/test_RunPlanVector.py::TestRunPlanVector::test_constructor - ValueError: invalid null reference in method 'RunPlanVector___ne__', argument 2 of type 'flamegpu::RunPlanVector co...
================================ 2 failed, 674 passed, 12 skipped in 727.95s (0:12:07) ================================
```

**Comment by Robadob** [comment]
Tbh the actual test looks a bit janky, it should probably just be a test that runplan constructor doesn't throw an exception, this equivalence doesn't really match the new/delete behaviour of the C test.

**Comment by ptheywood** [comment]
Can confirm this error is occuring under linux too, with Python 3.10 and Swig 4.2.1, however a local build using Swig 4.0.2 passess these tests successfully, implying it might be a change in swig? 

From users writing python, it should be possibel to compare to `None` without haveing to also check the type first, so it is something we should try and resolve so that the test (which has been unchanged in 3 years since it was originally written) passes.

The rc1 wheels from our wheel house which were built with 4.0.2 also still pass the test.

**Comment by Robadob** [comment]
Can't reproduce this today with SWIG 4.2.1 on Windows/Debug 🤷‍♂️

**Comment by Robadob** [comment]
Swig 4.3.0

Test passes (skipped test is operator+, test that should fail is constructor).

But there's some new deprecation warnings.

```pytest
(venv) C:\Users\Robadob\fgpu2\tests\python>py.test simulation/test_runplan.py
================================================= test session starts =================================================
platform win32 -- Python 3.12.3, pytest-8.3.3, pluggy-1.5.0
rootdir: C:\Users\Robadob\fgpu2\tests\python
collected 12 items

simulation\test_RunPlan.py .....s......                                                                          [100%]

================================================== warnings summary ===================================================
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type swigvarlink has no __module__ attribute

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
====================================== 11 passed, 1 skipped, 3 warnings in 0.60s ======================================
```

**Comment by ptheywood** [comment]
https://github.com/swig/swig/issues/2987 seems like the relevant issue discussion showing the behaviour we observed, which is now fixed in 4.3.0 (confirmed on linux too). 

Changing my non-idiomatic python from `!= None` to `is not None` does allow the test to pass with swig 4.2.1, which is a trivial fix so worth making, although 4.3.0 should still be preffered to 4.2.x? 

(also needs fixing in `test_RunPlanVector.py` and `test_cuda_ensemble.py`

---

I did also see the no `__module__` with 4.3.0, but they are not emitted with 4.2.1. It's targetted for fixing in swig in 4.4.0. 

From that thread it should only be visible to users if they invoke python with `-Walways`, or are inside pytest (which reports deprecation warnings by default unlike python itself). 

We could prevent our pytest suite from reporting these via a configuration file, equivalent to `python3 -m pytest -W ignore::DeprecationWarning ../tests/` (which does suppress them)? Though I'm also fine ignoring them.

**Comment by Robadob** [comment]
> [swig/swig#2987](https://github.com/swig/swig/issues/2987) seems like the relevant issue discussion showing the behaviour we observed, which is now fixed in 4.3.0 (confirmed on linux too).

Bah, i was searching `None` should have searched `operator==`.
